lib/Corres_UL.thy

```diff
@@ -1176,6 +1176,18 @@ lemma corres_move_asm:
 
 lemmas corres_cross_over_guard = corres_move_asm[rotated]
 
+lemma corres_cross_add_guard:
+  "\<lbrakk> \<And>s s'. \<lbrakk>(s,s') \<in> sr; P s; P' s'\<rbrakk> \<Longrightarrow> Q' s';
+     corres_underlying sr nf nf' r P (P' and Q') f g\<rbrakk>
+    \<Longrightarrow> corres_underlying sr nf nf' r P P' f g"
+  by (fastforce simp: corres_underlying_def)
+
+lemma corres_cross_add_abs_guard:
+  "\<lbrakk> \<And>s s'. \<lbrakk>(s,s') \<in> sr; P s; P' s'\<rbrakk> \<Longrightarrow> Q s;
+     corres_underlying sr nf nf' r (P and Q)  P' f g\<rbrakk>
+    \<Longrightarrow> corres_underlying sr nf nf' r P P' f g"
```

Comment:
ah, I only now noticed this, shouldn't the `⟹` align with the `⟦`?

Reply:
There are quite a few lemmas in this theory file with this same problem. Should I fix up just this one, or all of them?

Reply:
Well, I can at least fix it in the lemmas I've introduced in this PR. But the one above is old and wrong, and there are other instances of this elsewhere

Reply:
For this round, I'd stick to the ones you're adding, but we'd welcome a future PR that consolidates the others.